### PR TITLE
test(e2e): move stubs creation within the test itself and rewrite mocking features

### PR DESCRIPTION
**Pull request by CachedaCodes**
[EX-4533](https://searchbroker.atlassian.net/browse/EX-4533)

## Motivation and Context ##

To reduce complexity and have more explicit code when mocking API responses, the creation of the stub is now moved within the tests that require them instead of external files.

Also, a more natural language for the features regarding mocks. Avoiding abbreviations and words like "stub" or "dummy".

* NQ tests refactored
* Recommendations test also refactored slightly. It wasn't part of the scope of the task but was a minor enough change and was added for consistency

## How Has This Been Tested? ##

E2E tests should be working as they worked previously as this is a refactor.

### CODE REVIEW

**Review by pmareke — APPROVED** [review]
Nice job, well done!

**Review by davidmfempathy — APPROVED** [review]

File: packages/x-components/tests/e2e/cucumber/next-queries/next-queries.feature [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
+  Background:
+    Given a next queries API with a known response
+
   Scenario Outline: 1. Next query is clicked
-    Given next queries API should respond with mocked next queries
-    And   following config: hide session queries <hideSessionQueries>, requested items <maxItemsToRequest>, loadOnInit <loadOnInit>
+    Given following config: hide session queries <hideSessionQueries>, requested items <maxItemsToRequest>, loadOnInit <loadOnInit>
     When  "<query>" is searched
     Then  number of rows requested in "<request>" is <maxItemsToRequest>
     Then  at most <maxItemsToRequest> next queries are displayed
```
Comment (mavmaf) [inline packages/x-components/tests/e2e/cucumber/next-queries/next-queries.feature:10]: **And**  at most <maxItemsToRequest> next queries are displayed

File: packages/x-components/tests/e2e/cucumber/next-queries/next-queries.feature [hunk 2]
```diff
@@ -33,8 +34,7 @@ Feature: Next queries component
       | false              | 5                 | true       | playmobil | 0             | interceptedNextQueries |
 
   Scenario Outline: 3. Next queries persistence
-    Given next queries API should respond with mocked next queries
-    And   following config: hide session queries <hideSessionQueries>, requested items <maxItemsToRequest>, loadOnInit <loadOnInit>
+    Given following config: hide session queries <hideSessionQueries>, requested items <maxItemsToRequest>, loadOnInit <loadOnInit>
     When  "<query>" is searched
     Then  number of rows requested in "<request>" is <maxItemsToRequest>
     Then  related results are displayed
```
Comment (mavmaf) [inline packages/x-components/tests/e2e/cucumber/next-queries/next-queries.feature:40]: **And**  related results are displayed

File: packages/x-components/tests/e2e/cucumber/recommendations/recommendations.feature [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
 
+  Background:
+    Given a recommendations API with a known response
+
   Scenario Outline:  1. Recommendations are displayed
-    Given suggestions API should respond with mocked suggestions
-    And   following config: max items to store is <maxItemsToRequest>
+    Given following config: max items to store is <maxItemsToRequest>
     Then  number of rows requested in "<request>" is <maxItemsToRequest>
     Then  number of displayed recommendations are equal or less than <maxItemsToRequest>
```
Comment (mavmaf) [inline packages/x-components/tests/e2e/cucumber/recommendations/recommendations.feature:9]: **And**  number of displayed recommendations are equal or less than <maxItemsToRequest>

**Review by mavmaf — CHANGES_REQUESTED** [review]

**Pull request merged**